pkg/shares/compact_shares_test.go

```diff
@@ -112,6 +112,20 @@ func Test_processCompactShares(t *testing.T) {
 	}
 }
 
+func TestParseOutOfContextShares(t *testing.T) {
+	txs := testfactory.GenerateRandomlySizedTxs(1000, 150)
+	txShares, _, _, err := SplitTxs(txs)
+	require.NoError(t, err)
+
+	for i := 0; i < 1000; i++ {
+		start, length := testfactory.GetRandomSlice(len(txShares))
+
+		resTxs, err := ParseTxs(txShares[start : start+length])
+		require.NoError(t, err)
+		assert.True(t, testfactory.CheckSubArray(txs, resTxs))
```

Comment:
[blocking]
this is checking that the transactions returned were included, but I don't think its testing that all of the expected transactions were returned, correct? 

If so, then I think we _also_ need to also check for that. Perhaps we could use the shareRange returned somehow?

for example:

```go
        txShares, _, shareRanges, err := SplitTxs(txs)
	require.NoError(t, err)

	for key, r := range shareRanges {
		resTxs, err := ParseTxs(txShares[r.Start:r.End])
		require.NoError(t, err)
		has := false
		for _, tx := range resTxs {
			if tx.Key() == key {
				has = true
				break
			}
		}
		assert.True(t, has)
	}
}
```

Reply:
Yes, this is checking that the sub-list of transactions returned is a subarray of the original list of transactions. Added another test that covers shareRanges and checks a transaction in a shareRange is parsed correctly.